bjoern/request.c

```diff
@@ -205,7 +205,18 @@ on_message_complete(http_parser* parser)
   /* SERVER_NAME and SERVER_PORT */
   if (REQUEST->server_info->host) {
     _set_header(_SERVER_NAME, REQUEST->server_info->host);
-    _set_header(_SERVER_PORT, REQUEST->server_info->port);
+    if (REQUEST->server_info->port == Py_None) {
+      _set_header(_SERVER_PORT,
+                  PyUnicode_FromFormat(""));
+    } else {
+      _set_header(_SERVER_PORT,
+                  PyUnicode_FromFormat("%i", REQUEST->server_info->port));
+    }
+  }
+  /* SERVER_NAME is required */
+  else {
```

Comment:
In theory neither SERVER_NAME nor SERVER_PORT must be empty. Should a sane default of "localhost" be used? HTTP_HOST should take

Reply:
No, let's leave them empty. This is a case not covered by PEP 333, and we better break applications completely rather than sneaking in wrong values that cause hard-to-debug issues.